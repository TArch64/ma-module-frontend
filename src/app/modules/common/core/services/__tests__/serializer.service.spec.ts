import {SerializerService} from "../serializer.service";

const createService = () => new SerializerService()

describe('from json', () => {
    test('should parse json', () => {
        const service = createService();

        expect(service.fromJSON('{ "test": 123 }')).toEqual({ test: 123 });
    });

    test('should return null on invalid json', () => {
        const consoleError = jest.spyOn(console, 'error').mockImplementation();
        const service = createService();

        expect(service.fromJSON('{ "test": 1')).toEqual(null);
        consoleError.mockRestore();
    });
});

describe('to json', () => {
    test('should create json', () => {
        const service = createService();

        expect(service.toJSON({ test: 123 })).toMatchSnapshot()
    });
});
