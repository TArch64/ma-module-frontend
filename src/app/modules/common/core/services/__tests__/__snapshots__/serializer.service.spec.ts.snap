// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to json should create json 1`] = `"{\\"test\\":123}"`;
